shared/src/main/scala/mlscript/NewParser.scala

```diff
@@ -137,7 +137,10 @@ abstract class NewParser(origin: Origin, tokens: Ls[Stroken -> Loc], raiseFun: D
   
   private def cur(implicit l: Line, n: Name) = {
     if (dbg) printDbg(s"? ${n.value}\t\tinspects ${summarizeCur}    [at l.${l.value}]")
-    _cur
+    _cur.dropWhile(_._1 match {
+      case COMMENT(_) => true
+      case _ => false
+    })
```

Comment:
I think it would be better as:
```suggestion
    while (_cur match {
      case COMMENT(_) => true
      case _ => false
    }) consume
    _cur
```